### PR TITLE
Gometalinter script

### PR DESCRIPTION
**Pull request by mehmetalisavas**
Gometalinter does not check all lints
Enabled linters are;
- deadcode
- ineffassign
- structcheck
- gosimple
- misspell
- unconvert

### CODE REVIEW

File: go/src/koding/klientctl/mount.go [hunk 1]
```diff
@@ -605,7 +604,7 @@ func (c *MountCommand) cacheWithProgress(cacheReq req.Cache) (err error) {
 		bar.Set(p.Progress)
 
 		// TODO: Disable the callback here, so that it's impossible to double call
-		// the progress after competion - to avoid weird/bad UX and errors.
+		// the progress after competition - to avoid weird/bad UX and errors.
```
Comment (cihangir) [inline go/src/koding/klientctl/mount.go]: change all competition to completion in this PR :)

File: go/src/koding/remoteapi/client/collaboration/post_remote_api_collaboration_add_responses.go [hunk 2]
```diff
@@ -49,7 +49,7 @@ func NewPostRemoteAPICollaborationAddOK() *PostRemoteAPICollaborationAddOK {
 
 /*PostRemoteAPICollaborationAddOK handles this case with default header values.
 
-Request processed succesfully
```
Comment (cihangir) [inline go/src/koding/remoteapi/client/collaboration/post_remote_api_collaboration_add_responses.go]: https://github.com/koding/koding/blob/master/scripts/api-generator.coffee#L169

if you change here, all will be fixed

File: wercker.yml [hunk 3]
```diff
@@ -115,6 +115,9 @@ build:
         name: check social api code quality
         code: scripts/wercker/run-command scripts/check_socialapi_code_quality.sh
     - script:
+        name: check go codes with gometalinter
+        code: scripts/wercker/run-command scripts/gometalinter.sh
```
Comment (cihangir) [inline wercker.yml]: 💯

**Review by cihangir — CHANGES_REQUESTED** [review]
there are some small changes, socialapi part LGTM, overall looks good

**Comment by cihangir** [comment]
cc/ @ppknap please see enabled linters

**Comment by mehmetalisavas** [comment]
You can review & merge this PR. I will update config.json & move the flags into the config file.

File: go/src/koding/klientctl/mount.go (hunk 1 above)
Comment (mehmetalisavas) [inline go/src/koding/klientctl/mount.go]: 👍

File: go/src/koding/remoteapi/client/collaboration/post_remote_api_collaboration_add_responses.go (hunk 2 above)
Comment (mehmetalisavas) [inline go/src/koding/remoteapi/client/collaboration/post_remote_api_collaboration_add_responses.go]: 👍

**Comment by mehmetalisavas** [comment]
PTAL

**Review by cihangir — APPROVED** [review]

**Comment by mehmetalisavas** [comment]
@rjeczalik @ppknap  ping

**Review by ppknap — APPROVED** [review]
LGTM :+1:

**Comment by mehmetalisavas** [comment]
@rjeczalik  ping :)

**Comment by cihangir** [comment]
@rjeczalik please check when you have time

**Pull request merged**